From 30778275b098c2fb7517fe0882cdf8b26d4c3923 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 19 Sep 2014 12:29:34 -0400
Subject: [acpi] hotplug/pci: Eliminate acpiphp_dev_to_bridge()

Message-id: <1411129776-21656-60-git-send-email-prarit@redhat.com>
Patchwork-id: 96391
O-Subject: [RHEL7.1 PATCH BZ 1141846 59/61] ACPI / hotplug / PCI: Eliminate acpiphp_dev_to_bridge()
Bugzilla: 1141846
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Myron Stowe <mstowe@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1141846

commit 86f5f3ca49e3d20c1a5e83917b2c8b98a7c95506
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Tue Jul 15 22:03:14 2014 +0200

    ACPI / hotplug / PCI: Eliminate acpiphp_dev_to_bridge()

    Since acpiphp_dev_to_bridge() is only called by
    acpiphp_check_host_bridge(), move the code from it to that function
    directly which reduces the call chain depth and makes the code
    slightly easier to follow.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Acked-by: Bjorn Helgaas <bhelgaas@google.com>

Cc: Myron Stowe <mstowe@redhat.com>
Cc: Andy Gospodarek <agospoda@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/hotplug/acpiphp_glue.c b/drivers/pci/hotplug/acpiphp_glue.c
index af53580..89bfeb7 100644
--- a/drivers/pci/hotplug/acpiphp_glue.c
+++ b/drivers/pci/hotplug/acpiphp_glue.c
@@ -372,20 +372,6 @@ static acpi_status acpiphp_add_context(acpi_handle handle, u32 lvl, void *data,
  return AE_OK;
 }
 
-static struct acpiphp_bridge *acpiphp_dev_to_bridge(struct acpi_device *adev)
-{
- struct acpiphp_bridge *bridge = NULL;
-
- acpi_lock_hp_context();
- if (adev->hp) {
-  bridge = to_acpiphp_root_context(adev->hp)->root_bridge;
-  if (bridge)
-   get_bridge(bridge);
- }
- acpi_unlock_hp_context();
- return bridge;
-}
-
 static void cleanup_bridge(struct acpiphp_bridge *bridge)
 {
  struct acpiphp_slot *slot;
@@ -757,9 +743,15 @@ static void acpiphp_sanitize_bus(struct pci_bus *bus)
 
 void acpiphp_check_host_bridge(struct acpi_device *adev)
 {
- struct acpiphp_bridge *bridge;
+ struct acpiphp_bridge *bridge = NULL;
 
- bridge = acpiphp_dev_to_bridge(adev);
+ acpi_lock_hp_context();
+ if (adev->hp) {
+  bridge = to_acpiphp_root_context(adev->hp)->root_bridge;
+  if (bridge)
+   get_bridge(bridge);
+ }
+ acpi_unlock_hp_context();
  if (bridge) {
   pci_lock_rescan_remove();
 
-- 
1.7.1